impeller/toolkit/interop/impeller.h

```diff
@@ -0,0 +1,762 @@
+} ImpellerClipOperation;
+
+typedef enum ImpellerBlendMode {
+  kImpellerBlendModeClear,
+  kImpellerBlendModeSource,
+  kImpellerBlendModeDestination,
+  kImpellerBlendModeSourceOver,
+  kImpellerBlendModeDestinationOver,
+  kImpellerBlendModeSourceIn,
+  kImpellerBlendModeDestinationIn,
+  kImpellerBlendModeSourceOut,
+  kImpellerBlendModeDestinationOut,
+  kImpellerBlendModeSourceATop,
+  kImpellerBlendModeDestinationATop,
+  kImpellerBlendModeXor,
+  kImpellerBlendModePlus,
+  kImpellerBlendModeModulate,
+  kImpellerBlendModeScreen,
+  kImpellerBlendModeOverlay,
+  kImpellerBlendModeDarken,
+  kImpellerBlendModeLighten,
+  kImpellerBlendModeColorDodge,
+  kImpellerBlendModeColorBurn,
+  kImpellerBlendModeHardLight,
+  kImpellerBlendModeSoftLight,
+  kImpellerBlendModeDifference,
+  kImpellerBlendModeExclusion,
+  kImpellerBlendModeMultiply,
+  kImpellerBlendModeHue,
+  kImpellerBlendModeSaturation,
+  kImpellerBlendModeColor,
+  kImpellerBlendModeLuminosity,
+} ImpellerBlendMode;
+
+typedef enum ImpellerDrawStyle {
+  kImpellerDrawStyleFill,
+  kImpellerDrawStyleStroke,
+  kImpellerDrawStyleStrokeAndFill,
+} ImpellerDrawStyle;
+
+typedef enum ImpellerStrokeCap {
+  kImpellerStrokeCapButt,
+  kImpellerStrokeCapRound,
+  kImpellerStrokeCapSquare,
+} ImpellerStrokeCap;
+
+typedef enum ImpellerStrokeJoin {
+  kImpellerStrokeJoinMiter,
+  kImpellerStrokeJoinRound,
+  kImpellerStrokeJoinBevel,
+} ImpellerStrokeJoin;
+
+typedef enum ImpellerPixelFormat {
+  kImpellerPixelFormatRGBA8888,
+} ImpellerPixelFormat;
+
+typedef enum ImpellerTextureSampling {
+  kImpellerTextureSamplingNearestNeighbor,
+  kImpellerTextureSamplingLinear,
+} ImpellerTextureSampling;
+
+typedef enum ImpellerTileMode {
+  kImpellerTileModeClamp,
+  kImpellerTileModeRepeat,
+  kImpellerTileModeMirror,
+  kImpellerTileModeDecal,
+} ImpellerTileMode;
+
+typedef enum ImpellerBlurStyle {
+  kImpellerBlurStyleNormal,
+  kImpellerBlurStyleSolid,
+  kImpellerBlurStyleOuter,
+  kImpellerBlurStyleInner,
+} ImpellerBlurStyle;
+
+//------------------------------------------------------------------------------
+// Non-opaque structs
+//------------------------------------------------------------------------------
+typedef struct ImpellerRect {
+  float x;
+  float y;
+  float width;
+  float height;
+} ImpellerRect;
+
+typedef struct ImpellerPoint {
+  float x;
+  float y;
+} ImpellerPoint;
+
+typedef struct ImpellerSize {
+  float width;
+  float height;
+} ImpellerSize;
+
+typedef struct ImpellerISize {
+  int64_t width;
+  int64_t height;
+} ImpellerISize;
+
+typedef struct ImpellerMatrix {
+  float m[16];
+} ImpellerMatrix;
+
+typedef struct ImpellerColorMatrix {
+  float m[20];
+} ImpellerColorMatrix;
+
+typedef struct ImpellerRoundingRadii {
+  ImpellerPoint top_left;
+  ImpellerPoint bottom_left;
+  ImpellerPoint top_right;
+  ImpellerPoint bottom_right;
+} ImpellerRoundingRadii;
+
+typedef struct ImpellerColor {
+  float red;
+  float green;
+  float blue;
+  float alpha;
```

Comment:
These should have a color space attached, which even if we only support srgb on gles will save some trouble in the future.

Reply:
I was thinking of expanding the struct later since we didn't guarantee stability anyway. But yeah, I can add it now just the same.

Reply:
Okie dokie. I added the colorspace to and patched the implementation for gradients and such to pass that information along.